src/main/scala/mesosphere/marathon/api/v2/PodsResource.scala

```diff
@@ -31,4 +50,207 @@ class PodsResource @Inject() (
   def capability(@Context req: HttpServletRequest): Response = authenticated(req) { _ =>
     ok()
   }
+
+  // TODO(jdef) force parameter? probably since this supports deployment; but what does it actually
+  // mean to "force" a create?
+  @POST @Timed
+  def create(
+    body: Array[Byte],
+    @DefaultValue("false")@QueryParam("force") force: Boolean,
+    @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+
+    withValid(unmarshalJson[PodDef](decodeBytes(body, req))) { podDef =>
+
+      val pod = PodDefinition(withDefaults(podDef, podDefaults)).withCanonizedIds()
+
+      withAuthorization(CreateRunSpec, pod) {
+        val (createdPod, deploymentId) = podSystem.create(pod, force)
+        Events.maybePost(PodEvent(req.getRemoteAddr, req.getRequestURI, PodEvent.Created))
+
+        val builder = Response
+          .created(new URI(createdPod.id.toString))
+          .entity(marshalJson(createdPod.asPodDef))
+
+        deploymentId.foreach(id => builder.header(DEPLOYMENT_ID_HEADER, id))
+        builder.build()
+      }
+    }(createPodValidator)
+  }
+
+  @GET @Timed
+  def findAll(@Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+
+    val authSelector = Selector { pod =>
+      isAuthorized(ViewRunSpec, pod)
+    }
+    val pods: Iterable[PodDefinition] = podSystem.findAll(authSelector)
+    ok(marshalJson(pods.map(_.asPodDef)))
+  }
+
+  @GET @Timed @Path("""{id:.+}""")
+  def find(
+    @PathParam("id") id: String,
+    @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+
+    withValid(id) { _ =>
+      val pod: PodDefinition = podSystem.find(id)
+      withAuthorization(ViewRunSpec, pod) {
+        ok(marshalJson(pod.asPodDef))
+      }
+    }(PodsValidation.idValidator)
+  }
+
+  @PUT @Timed @Path("""{id:.+}""")
+  def update(
+    @PathParam("id") id: String,
+    body: Array[Byte],
+    @DefaultValue("false")@QueryParam("force") force: Boolean,
+    @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+
+    withValid(unmarshalJson[PodDef](decodeBytes(body, req))) { podDef =>
+
+      // podDef.id has just been validated, requiring equality for id validates it
+      require(id == podDef.id)
```

Comment:
More like Bad Input or something. Because the ID param didn't match the ID supplied with the json